File: HackerRank/diagonalDifference/difference.java
import java.util.Scanner;

public class difference {
  public static void main(String[] args) {
    Scanner this.keyboard = new Scanner(system.in);
    this.getInput();
    a = new DifferenceArray(this.arrays);
    System.out.println(a.diagonalDifference());
  }
  private void getInput(){
    int number = Integer.parseInt(this.keyboard.nextLine());
    int[][] this.arrays = new int[number][number]
    for (int index=0; index < number; index++) {
      String[] row = this.keyboard.nextLine().split("\\s+");
      for (int index2=0; index2<row.length(); index2++) {
        this.arrays[index][index2] = Integer.parseInt(row[index2]);
      }
    }
  }
}
private class DifferenceArray {
  DifferenceArray(int[][] arrays){
    this.arrays = arrays;
  }
  private int leftDiagonalSum(){
    int outsum = 0;
    int row    = 0;
    int column = this.arrays.length - 1;
    while (row < arrays.length){
      outsum += arrays[row][column];
      row    ++;
      column --;
    }
    return outsum;
  }
  private int rightDiagonalSum(){
    int outsum = 0;
    for (int index=0; index<this.arrays.length; index ++) {
      outsum += this.arrays[index][index];
    }
    return outsum;
  }
  public int diagonalDifference(){
    return Math.abs(this.rightDiagonalSum() - this.leftDiagonalSum());
  }
}

File: HackerRank/diagonalDifference/difference.py
from typing import List
"""The absolute value of the difference between the diagonals of a 2D array.

input should be:
Width/Height of the array on the first input line
any subsequent line should contain the space-separated values.
"""


def right_diagonal_sum(arrays: List[List[int]]) -> int:
    """Sum the right diagonal of the given 2d array."""
    outsum = 0
    for index in range(len(arrays)):
        outsum += arrays[index][index]
    return outsum


def left_diagonal_sum(arrays: List[List[int]]) -> int:
    """Sum the left diagonal of the given 2d array."""
    outsum = 0
    row = 0
    column = len(arrays) - 1
    while row < len(arrays):
        outsum += arrays[row][column]
        column -= 1
        row    += 1
    return outsum


def diagonal_difference(arrays: List[List[int]]) -> int:
    """Get the absolute value of the difference between the diagonals.

    @param arrays:  must be a sqare matrix - i.e. an array of arrays where the
                    number of arrays is the same as the number of values in
                    each array.
    """
    return abs(right_diagonal_sum(arrays) - left_diagonal_sum(arrays))


def main():
    rows = int(input("Number of arrays to be specified."))
    a = [
        int(element) for element in [
            input(f"Array #{row}: ").split(' ') for row in range(rows)
        ]
    ]
    print(diagonal_difference(a))
